### PR TITLE
Make user-data-permission-fix optional

### PR DESCRIPTION
**Pull request by nerdinand**
### Motivation and context

Deploying CVAT to OpenShift (see also #3992), we've run into problems with the `user-data-permission-fix` initContainer. OpenShift seems to be stricter in what you can do permission-wise. To mitigate this, we propose making the permission fix optional. Along with adding `RUN /bin/chmod -R 777 ${HOME}` to the Dockerfile, this works for us.

### How has this been tested?

Helm chart tested manually

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- ~~[ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- ~~[ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nerdinand** [comment]
Maybe a better solution would be to add `RUN /bin/chmod -R 777 ${HOME}` to the Dockerfile in the first place, then this initContainer might become obsolete altogether. Thoughts?

**Comment by nmanovic** [comment]
@azhavoro , @ActiveChooN , could you please look?

**Comment by ActiveChooN** [comment]
> Maybe a better solution would be to add `RUN /bin/chmod -R 777 ${HOME}` to the Dockerfile in the first place, then this initContainer might become obsolete altogether. Thoughts?

If this fix can be applied in the build stage it looks like a better solution.

**Comment by nmanovic** [comment]
@azhavoro , could you please review and share your opinion? Should we merge the PR?

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Data sorting option (<https://github.com/openvinotoolkit/cvat/pull/3937>)
 - Options to change font size & position of text labels on the canvas (<https://github.com/openvinotoolkit/cvat/pull/3972>)
 - Add "tag" return type for automatic annotation in Nuclio (<https://github.com/openvinotoolkit/cvat/pull/3896>)
+- Helm chart: Make user-data-permission-fix optional (https://github.com/openvinotoolkit/cvat/pull/3994)
```
Comment (azhavoro) [inline CHANGELOG.md:24]:
```suggestion
- Helm chart: Make user-data-permission-fix optional (<https://github.com/openvinotoolkit/cvat/pull/3994>)
```

**Review by azhavoro — APPROVED** [review]
LGTM

**Review by nmanovic — APPROVED** [review]
@nerdinand , thanks for the contribution.

**Comment by sunyd1006** [comment]
Maybe securityContext is a better solution.
https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod

**Pull request merged**